---
title: "04-Bismark-Mox"
output: html_document
---

Bismark was run on Mox

```
#!/bin/bash
## Job Name
#SBATCH --job-name=bismark
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes (We only get 1, so this is fixed)
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-100:00:00
## Memory per node
#SBATCH --mem=100G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --workdir=/gscratch/srlab/sr320/analyses/0618b
 
 
 
 
source /gscratch/srlab/programs/scripts/paths.sh
 
 
/gscratch/srlab/programs/Bismark-0.19.0/bismark_genome_preparation \
--verbose \
/gscratch/srlab/sr320/data/olurida-genomes/v081
 
 
 
find /gscratch/srlab/sr320/data/olurida-bs/[1-8]_*.gz \
| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /gscratch/srlab/programs/Bismark-0.19.0/bismark \
--path_to_bowtie /gscratch/srlab/programs/bowtie2-2.1.0 \
--score_min L,0,-0.9 \
-genome /gscratch/srlab/sr320/data/olurida-genomes/v081 \
-p 28 \
/gscratch/srlab/sr320/data/olurida-bs/{}_L001_R1_001.fastq.gz
 
 
 
/gscratch/srlab/programs/Bismark-0.19.0/deduplicate_bismark \
--bam -s \
/gscratch/srlab/sr320/analyses/0618b/*.bam
 
 
/gscratch/srlab/programs/Bismark-0.19.0/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
/gscratch/srlab/sr320/analyses/0618b/*deduplicated.bam
```

Output is available @
http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/


```
1_ATCACG_L001_R1_001_bismark_bt2.bam
1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.bam
1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
1_ATCACG_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
1_ATCACG_L001_R1_001_bismark_bt2.deduplication_report.txt
1_ATCACG_L001_R1_001_bismark_bt2_SE_report.txt
2_CGATGT_L001_R1_001_bismark_bt2.bam
2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.bam
2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
2_CGATGT_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
2_CGATGT_L001_R1_001_bismark_bt2.deduplication_report.txt
2_CGATGT_L001_R1_001_bismark_bt2_SE_report.txt
3_TTAGGC_L001_R1_001_bismark_bt2.bam
3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.bam
3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
3_TTAGGC_L001_R1_001_bismark_bt2.deduplication_report.txt
3_TTAGGC_L001_R1_001_bismark_bt2_SE_report.txt
4_TGACCA_L001_R1_001_bismark_bt2.bam
4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.bam
4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
4_TGACCA_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
4_TGACCA_L001_R1_001_bismark_bt2.deduplication_report.txt
4_TGACCA_L001_R1_001_bismark_bt2_SE_report.txt
5_ACAGTG_L001_R1_001_bismark_bt2.bam
5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.bam
5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
5_ACAGTG_L001_R1_001_bismark_bt2.deduplication_report.txt
5_ACAGTG_L001_R1_001_bismark_bt2_SE_report.txt
6_GCCAAT_L001_R1_001_bismark_bt2.bam
6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.bam
6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
6_GCCAAT_L001_R1_001_bismark_bt2.deduplication_report.txt
6_GCCAAT_L001_R1_001_bismark_bt2_SE_report.txt
7_CAGATC_L001_R1_001_bismark_bt2.bam
7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.bam
7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
7_CAGATC_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
7_CAGATC_L001_R1_001_bismark_bt2.deduplication_report.txt
7_CAGATC_L001_R1_001_bismark_bt2_SE_report.txt
8_ACTTGA_L001_R1_001_bismark_bt2.bam
8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.bam
8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.bedGraph.gz
8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.bismark.cov.gz
8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.M-bias.txt
8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated_splitting_report.txt
8_ACTTGA_L001_R1_001_bismark_bt2.deduplication_report.txt
8_ACTTGA_L001_R1_001_bismark_bt2_SE_report.txt
CHG_OB_1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OB_8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.txt
CHG_OT_8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OB_8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.txt
CHH_OT_8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OB_8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.txt
CpG_OT_8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.txt
slurm-191972.out

```

# Running Methylkit

```{r setup, include = FALSE }
library(methylKit)
```


## Downloading de-duplicated bams.

```
1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.bam  5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.bam
2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.bam  6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.bam
3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.bam  7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.bam
4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.bam  8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.bam
```




```{bash}
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0618b/8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.bam

```


```{bash}
find ../data/*deduplicated.bam \
| xargs basename -s _L001_R1_001_bismark_bt2.deduplicated.bam | xargs -I{} /home/shared/samtools-1.8/samtools \
sort ../data/{}_L001_R1_001_bismark_bt2.deduplicated.bam -o ../data/{}_dedup.sorted.bam
```







```{r}
file.list=list(  '../data/1_ATCACG_dedup.sorted.bam',
                 '../data/2_CGATGT_dedup.sorted.bam',
                 '../data/3_TTAGGC_dedup.sorted.bam',
                 '../data/4_TGACCA_dedup.sorted.bam',
                 '../data/5_ACAGTG_dedup.sorted.bam',
                 '../data/6_GCCAAT_dedup.sorted.bam',
                 '../data/7_CAGATC_dedup.sorted.bam',
                 '../data/8_ACTTGA_dedup.sorted.bam'
)
```



```{r}
myobj = processBismarkAln(location = file.list, sample.id = list("1","2","3","4","5","6","7","8"), assembly = "v081", read.context="CpG", mincov=10, treatment = c(0,0,0,0,1,1,1,1))
```





