---
title: "06-Bismark-Mox"
output: html_document
---

Running Bismark with --score_min L,0,-0.6 
Note it took 8 hours.


```
#!/bin/bash
## Job Name
#SBATCH --job-name=bismark
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes (We only get 1, so this is fixed)
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-100:00:00
## Memory per node
#SBATCH --mem=100G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --workdir=/gscratch/srlab/sr320/analyses/0620b




source /gscratch/srlab/programs/scripts/paths.sh





find /gscratch/srlab/sr320/data/olurida-bs/[1-8]_*.gz \
| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /gscratch/srlab/programs/Bismark-0.19.0/bismark \
--path_to_bowtie /gscratch/srlab/programs/bowtie2-2.1.0 \
-genome /gscratch/srlab/sr320/data/olurida-genomes/v081 \
--score_min L,0,-0.6 \
-p 28 \
/gscratch/srlab/sr320/data/olurida-bs/{}_L001_R1_001.fastq.gz



/gscratch/srlab/programs/Bismark-0.19.0/deduplicate_bismark \
--bam -s \
/gscratch/srlab/sr320/analyses/0620b/*.bam


/gscratch/srlab/programs/Bismark-0.19.0/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
/gscratch/srlab/sr320/analyses/0620b/*deduplicated.bam

```


output
http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/



```
[sr320@mox2 0620b]$ cat *SE_report.txt | grep "Mapping"
Mapping efficiency:	54.1%
Mapping efficiency:	53.3%
Mapping efficiency:	53.9%
Mapping efficiency:	54.2%
Mapping efficiency:	53.3%
Mapping efficiency:	52.8%
Mapping efficiency:	53.7%
Mapping efficiency:	52.8%
```



## Downloading de-duplicated bams.



```{bash, eval = FALSE}
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/1_ATCACG_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/2_CGATGT_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/3_TTAGGC_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/4_TGACCA_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/5_ACAGTG_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/6_GCCAAT_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/7_CAGATC_L001_R1_001_bismark_bt2.deduplicated.bam
curl -O  http://gannet.fish.washington.edu/seashell/bu-mox/analyses/0620b/8_ACTTGA_L001_R1_001_bismark_bt2.deduplicated.bam

```





```{bash, eval = FALSE}
find ../data_06/*deduplicated.bam \
| xargs basename -s _L001_R1_001_bismark_bt2.deduplicated.bam | xargs -I{} /home/shared/samtools-1.8/samtools \
sort ../data_06/{}_L001_R1_001_bismark_bt2.deduplicated.bam -o ../data_06/{}_dedup.sorted.bam
```
```{r}
file.list_06=list(  '../data_06/1_ATCACG_dedup.sorted.bam',
                 '../data_06/2_CGATGT_dedup.sorted.bam',
                 '../data_06/3_TTAGGC_dedup.sorted.bam',
                 '../data_06/4_TGACCA_dedup.sorted.bam',
                 '../data_06/5_ACAGTG_dedup.sorted.bam',
                 '../data_06/6_GCCAAT_dedup.sorted.bam',
                 '../data_06/7_CAGATC_dedup.sorted.bam',
                 '../data_06/8_ACTTGA_dedup.sorted.bam'
)
```



```{r, eval = FALSE}
myobj_06_10x = processBismarkAln(location = file.list_06, sample.id = list("1","2","3","4","5","6","7","8"), assembly = "v081", read.context="CpG", mincov=10, treatment = c(0,0,0,0,1,1,1,1))
```

```{r, eval = FALSE}
save(myobj_06_10x, file = "../analyses/myobj_06_10x")

```


```{r}
load("../analyses/myobj_10x")
```




```{r setup, include = FALSE }
library(methylKit)
```




```{r}
meth_06_10x <- unite(myobj_06_10x)
```

```{r}
clusterSamples(meth_06_10x, dist="correlation", method="ward", plot=TRUE)

```


```{r}
myobj_06_3x = processBismarkAln(location = file.list_06, sample.id = list("1","2","3","4","5","6","7","8"), assembly = "v081", read.context="CpG", mincov=3, treatment = c(0,0,0,0,1,1,1,1))
```







```{r}
meth_06_3x=unite(myobj_06_3x)
```

```{r}
clusterSamples(meth_06_3x, dist="correlation", method="ward", plot=TRUE)

```